### PR TITLE
Issue #3904: Split and Organize Checkstyle inputs by Test for Separat…

### PR DESCRIPTION
**Pull request by sagar-shah94**
Issue: #3904

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void setUp() {
     @Override
     protected String getPath(String filename) throws IOException {
         return super.getPath("checks" + File.separator
-                + "whitespace" + File.separator + filename);
+                + "SeparatorWrap" + File.separator + filename);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Package name should be lowercase.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public void testDot()
         final String[] expected = {
             "31:10: " + getCheckMessage(MSG_LINE_NEW, "."),
         };
-        verify(checkConfig, getPath("InputSeparatorWrap.java"), expected);
+        verify(checkConfig, getPath("InputSeparatorWrapAfter3904.java"), expected);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Do not make input file names cryptic. Please rename the file to clearly state its purpose.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 3]
```diff
@@ -70,7 +70,7 @@ public void testComma() throws Exception {
         final String[] expected = {
             "39:17: " + getCheckMessage(MSG_LINE_PREVIOUS, ","),
         };
-        verify(checkConfig, getPath("InputSeparatorWrap.java"), expected);
+        verify(checkConfig, getPath("InputSeparatorWrapAfter3904.java"), expected);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Separate file for each test. Rename it to avoid misunderstanding. File name should contain short information about its purpose.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 4]
```diff
@@ -91,7 +91,7 @@ public void testInvalidOption() throws Exception {
         try {
             final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
 
-            verify(checkConfig, getPath("InputSeparatorWrap.java"), expected);
+            verify(checkConfig, getPath("InputSeparatorWrapAfter3904.java"), expected);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Separate file. Rename it to avoid misunderstanding. File name should contain short information about its purpose.

Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: if i make different files and put those files in same folder so i will not be able to make all interfaces like foo,bar etc in every file

Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @sagar-shah94 
Why you cannot rename the interfaces. You are free to rename any inner language structures. The main idea is to make it easy for future contributor to add new tests and understand your test. That is why it is better to split input files to shorten the context of the test.

Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: ok sir i will change interfaces name and push code

Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Are following files names fine?
InputSeparatorWrapForTestDot.java
InputSeparatorForTestComma.java
InputSeparatorWrapForInvalidOption.java
@MEZk

**Comment by sagar-shah94** [comment]
@MEZk made changes please verify and apporve

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 5]
```diff
@@ -49,7 +49,7 @@ public void setUp() {
     @Override
     protected String getPath(String filename) throws IOException {
         return super.getPath("checks" + File.separator
-                + "whitespace" + File.separator + filename);
+                + "separatorWrap" + File.separator + filename);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: In Java world packages should never have uppercase letters.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 6]
```diff
@@ -60,7 +60,7 @@ public void testDot()
         final String[] expected = {
             "31:10: " + getCheckMessage(MSG_LINE_NEW, "."),
         };
-        verify(checkConfig, getPath("InputSeparatorWrap.java"), expected);
+        verify(checkConfig, getPath("InputSeparatorWrapForTestDot.java"), expected);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: InputSeparatorWrapForTestDot does not need to contain examples for COMMA. Only for DOTS as specified in tokens set.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 7]
```diff
@@ -70,7 +70,7 @@ public void testComma() throws Exception {
         final String[] expected = {
             "39:17: " + getCheckMessage(MSG_LINE_PREVIOUS, ","),
         };
-        verify(checkConfig, getPath("InputSeparatorWrap.java"), expected);
+        verify(checkConfig, getPath("InputSeparatorWrapForTestComma.java"), expected);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: InputSeparatorWrapForTestComma does not need to contain examples for DOT. Only for COMMAS as specified in tokens set.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 8]
```diff
@@ -91,7 +91,7 @@ public void testInvalidOption() throws Exception {
         try {
             final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
 
-            verify(checkConfig, getPath("InputSeparatorWrap.java"), expected);
+            verify(checkConfig, getPath("InputSeparatorWrapForInvalidOption.java"), expected);
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: InputSeparatorWrapForInvalidOption can contain an empty class. The main purpose of the test is to check that passing ivalid option will throw exception.

**Comment by sagar-shah94** [comment]
@MEZk made changes ,please verify.

**Comment by MEZk** [comment]
@rnveach @romani  
Changes look OK to me. Please make final decision.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 9]
```diff
@@ -49,7 +49,7 @@ public void setUp() {
     @Override
     protected String getPath(String filename) throws IOException {
         return super.getPath("checks" + File.separator
-                + "whitespace" + File.separator + filename);
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Why did we remove `whitespace`? Expected directory structure should be `.../checks/whitespace/separatorwrap/...`

Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @rnveach 
Oh, yes. You are right. My fault. SeparatorWrap is a part of the checks which are in whitespace group.
http://checkstyle.sourceforge.net/config_whitespace.html

Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: should i move all files in whitespace/separatorwrap?

Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: yes as in @rnveach's example

**Comment by sagar-shah94** [comment]
@rnveach @MEZk done.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 10]
```diff
@@ -58,19 +58,19 @@ public void testDot()
         checkConfig.addAttribute("option", "NL");
         checkConfig.addAttribute("tokens", "DOT");
         final String[] expected = {
-            "31:10: " + getCheckMessage(MSG_LINE_NEW, "."),
+            "29:10: " + getCheckMessage(MSG_LINE_NEW, "."),
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Why did line number change if input was copied? We aren't expecting the input to change except for package.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java [hunk 11]
```diff
@@ -58,19 +58,19 @@ public void testDot()
...
+        verify(checkConfig, getPath("InputSeparatorWrapForTestDot.java"), expected);
     }
 
     @Test
     public void testComma() throws Exception {
         checkConfig.addAttribute("option", "EOL");
         checkConfig.addAttribute("tokens", "COMMA");
         final String[] expected = {
-            "39:17: " + getCheckMessage(MSG_LINE_PREVIOUS, ","),
+            "35:17: " + getCheckMessage(MSG_LINE_PREVIOUS, ","),
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: same here.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 10 above)
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @rnveach 
Input was not copied. It was changed in order to become more specific to test content. For example, if we test NL option the input file should contains positive and negative test cases with comments for new line. This will make the input file shorter and it content will become more readable. We already have input files which are hard to read because they contain to many unrelated code examples. I think that each input file should be as short as possible, each UT too.

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: [Issue says `copy`](https://github.com/checkstyle/checkstyle/issues/3904#issue-211755288). I think we should stick with copy as trimming input requires reviewing what was changed to make sure something necessary wasn't accidentally removed. Violation cases are hard to miss, but good cases are easy to miss as there is no verification in test. Since file is copied and then changed in 1 commit, we can't see in history what was changed.
@MEZk I am fine with trimming input, but it should be done outside this issue and most likely after the parent issue for this is finished as we have alot of tests to go through.
@romani can override this, but I stand by that we should only be doing copies for now since we are overloaded with PRs and GSoC.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 11 above)
Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @rnveach so i should just copy and paste same content in 3 files?

Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @rnveach @romani what am i supposed to do ?

Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @sagar-shah94 
As was discussed above (https://github.com/checkstyle/checkstyle/pull/3915#discussion_r104325446)  it is better to to make changes to input files with small steps. So, for this issue just copy-pasting inout files will be enough. Sorry, that I forced you to make unnecessary changes.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 10 above)
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @rnveach 
Agree. Lest postpone my idea till we resolve all issues related to splitting of input files.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 11 above)
Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @MEZk no problem sir .i will copy paste and let u know

**Comment by romani** [comment]
@sagar-shah94 , please reply to all our comments as "done" or ...... .
Some items are raised but not addressed by you, I will do review when you resolve all point from existing reviewers.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 9 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: This is done.

**Comment by sagar-shah94** [comment]
@MEZk I had to copy and paste same code in 3 separate file .Right?

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 9 above)
Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 8 above)
Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 7 above)
Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 6 above)
Comment (sagar-shah94) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: Done

**Comment by sagar-shah94** [comment]
@romani it was decided in last that i have to copy and paste same code  in 3 separate file instead of keeping specific examples in files.
@MEZk asked me to do that way.

**Comment by MEZk** [comment]
@romani @sagar-shah94 
It was asked in original issue:
> do a copy of all test input files to separate folder for SeparatorWrap .
Not any other Check should use new inputs.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java (hunk 4 above)
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SeparatorWrapCheckTest.java]: @sagar-shah94
InputSeparatorWrapDotAndNewLineOption
InputSeparatorWrapCommaAndEndOfLineOption
InputSeparatorWrapInvalidOption

**Comment by sagar-shah94** [comment]
@MEZk what other changes required?i think i have addressed all changes
